tests/monad.cpp

```diff
@@ -17,7 +22,33 @@ static auto unsafe_divide_4_by(double val) {
     return 4.0 / val;
 }
 
-TEST_CASE("rsl::mbind") {
+template <typename T>
+using result_t = tl::expected<T, std::string>;
+
+static result_t<double> divide(double x, double y) {
+    if (y == 0) {
+        return tl::make_unexpected(std::string{"divide by 0"});
+    }
+    return x / y;
+}
+
+static result_t<double> multiply(double x, double y) { return x * y; }
+
+static result_t<double> divide_3(double x) { return divide(3, x); }
+
+static result_t<double> multiply_3(double x) { return multiply(3, x); }
+
+template <typename OUT, typename IN>
+static result_t<OUT> to(IN in) {
+    auto const out = static_cast<OUT>(in);
+    auto const check = static_cast<IN>(out);
+    if (check != in) {
+        return tl::make_unexpected(std::string{"no worky casty"});
+    }
```

Comment:
```suggestion
    if (check != in)
        return tl::make_unexpected(std::string{"no worky casty"});
```
I just realized you probably include these unnecessary braces because Rust requires them.

Reply:
I do it by habit